src/common/utils/validators.py

```diff
@@ -200,3 +200,4 @@ def _validate_list(
             _validate_primitive_attribute(attributeDefinition, item, f"{key}.{i}")
         else:
             _validate_complex_attribute(attributeDefinition, item, get_blueprint, f"{key}.{i}", implementation_mode)
+            validate_entity_against_self(entity=item, get_blueprint=get_blueprint, key=f"{key}.{i}")
```

Comment:
there was a small bug with validation of the content list for packages.

Since the Package blueprint defines content to be a list of objects, no validation inside _validate_complex_attribute() was made. To check that the acutal entity is correct, I needed to add a call to validate_entity_against_self()

Reply:
was able to move the validation into _validate_complex_attribute, but had to add a new if check :)